### PR TITLE
MODINVOICE-103 Adjustment Total Calculation is wrong

### PR DESCRIPTION
**Pull request by javalkar-varun**
## Purpose

https://issues.folio.org/browse/MODINVOICE-103

## Approach

- If there are no invoice lines then adjustmentsTotal = sum of all invoice adjustments
- If lines are present then adjustmentsTotal = notProratedInvoiceAdjustments + sum of invoiceLines adjustmentsTotal
- Update unit tests

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/test/java/org/folio/rest/impl/InvoicesProratedAdjustmentsTest.java [hunk 1]
```diff
@@ -123,7 +123,7 @@ public void testUpdateInvoiceWithOneLineAddingAdjustment(Adjustment.Prorate pror
      * "Amount" - this is the same as adjustment value
      * "Percentage" - this is the percentage of subTotal i.e. 15% of 25$ = 3.75$
      */
-    double expectedAdjTotal = (type == AMOUNT) ? 25d : 13.75d;
+    double expectedAdjTotal = (type == AMOUNT) ? 65d : 31.25d;
```
Comment (pautau-s) [inline src/test/java/org/folio/rest/impl/InvoicesProratedAdjustmentsTest.java]: Invoice has prorated adjustment with value 15. Line has not prorated amount adjustment with value 10$ and subTotal 25$.
When prorated adjustment type is amount `expectedAdjTotal` should be  - 10.0 (line not prorated) + 15.0 (invoice prorated) = 25.
If prorated adjustment type is percentage `expectedAdjTotal` should be  - 10.0 (line not prorated) + 15 * 25 / 100 = 13.75

**Comment by javalkar-varun** [comment]
@pautau-s - So you updated an invoice by adding 2 lines? There is existing unit test with this use-case so I was thinking it is covered - https://github.com/folio-org/mod-invoice/blob/master/src/test/java/org/folio/rest/impl/InvoicesProratedAdjustmentsTest.java#L979 Information on how you tested might be helpful such as curl calls etc.

**Comment by pautau-s** [comment]
@javalkar-varun This test covers case when line already in storage.
See gif attached to jira.

**Comment by javalkar-varun** [comment]
This story was created only for the wrong total calculation when there are no lines which is covered. It seems like there are more issues with the total calculation which are cascading effects of this bug. We should merge this PR first and during testing, if I find more issues, will create new story. Also, I'm spending more time on this than expected.

**Pull request merged**